### ISSUE TITLE
4.0.4 broken with TypeScript strict mode

### ISSUE THREAD

**Issue by qtiki** [pixijs/sound#165]
Trying to import v4.0.4 in a project that has TypeScript strict mode enabled:
```
import * as PIXISound from '@pixi/sound';
```
causes errors like this:
```
node_modules/@pixi/sound/dist/pixi-sound.d.ts:384:5 - error TS2416: Property 'init' in type 'HTMLAudioInstance' is not assignable to the same property in base type 'IMediaInstance'.
  Type '(media: HTMLAudioMedia) => void' is not assignable to type '(parent: IMedia) => void'.
    Types of parameters 'media' and 'parent' are incompatible.
      Type 'IMedia' is missing the following properties from type 'HTMLAudioMedia': parent, _source, source, eventNames, and 9 more.

384     init(media: HTMLAudioMedia): void;
        ~~~~
```
The errors look a bit mysterious but I traced it down to the more strict event typings added in a0523318fc1c1c130da7c98f699e24ab8c13a948.

Just to be clear: v4.0.3 works with strict mode

**Comment by qtiki** [comment]
These errors appear when compiling the package if you enable `strictNullChecks`. Naturally that will also bring up a bunch of other errors too but still it might be helpful for narrowing down this issue. Some of those errors might be actual problems though so it probably wouldn't hurt to make the whole package compatible with strict null checks.

**Comment by qtiki** [comment]
I did a PR for this as it's blocking us from upgrading to 4.0.5 to get the fix for #169 which we need.
